plugins/by-name/copilot-lua/default.nix

```diff
@@ -1,172 +1,224 @@
-              Each value can be either a boolean or a lua function that returns a boolean.
-
-              Example:
-              ```nix
-                {
-                  markdown = true; # overrides default
-                  terraform = false; # disallow specific filetype
-                  sh.__raw = \'\'
-                    function ()
-                      if string.match(vim.fs.basename(vim.api.nvim_buf_get_name(0)), '^%.env.*') then
-                        -- disable for .env files
-                        return false
-                      end
-                      return true
-                    end
-                  \'\';
-                }
-              ```
-
-              The key `"*"` can be used to disable the default configuration.
-              Example:
-              ```nix
-                {
-                  javascript = true; # allow specific filetype
-                  typescript = true; # allow specific filetype
-                  "*" = false; # disable for all other filetypes and ignore default `filetypes`
-                }
-              ```
-            '';
-
-        copilotNodeCommand = mkOption {
-          type = types.str;
-          default = "${pkgs.nodejs-18_x}/bin/node";
-          description = ''
-            Use this field to provide the path to a specific node version such as one installed by
-            `nvm`.
-            Node.js version must be 16.x or newer.
-          '';
-        };
-
-        serverOptsOverrides = helpers.defaultNullOpts.mkAttrsOf' {
-          type = types.anything;
-          pluginDefault = { };
-          description = ''
-            Override copilot lsp client `settings`.
-            The settings field is where you can set the values of the options defined in
-            https://github.com/zbirenbaum/copilot.lua/blob/master/SettingsOpts.md.
-            These options are specific to the copilot lsp and can be used to customize its behavior.
-
-            Ensure that the `name` field is not overridden as is is used for efficiency reasons in
-            numerous checks to verify copilot is actually running.
-
-            See `:h vim.lsp.start_client` for list of options.
-          '';
-          example = {
-            trace = "verbose";
-            settings = {
-              advanced = {
-                listCount = 10; # number of completions for panel
-                inlineSuggestCount = 3; # number of completions for getCompletions
-              };
-            };
-          };
-        };
+  settingsExample = {
+    panel = {
+      enabled = true;
+      auto_refresh = true;
+    };
+    suggestion = {
+      enabled = true;
+      auto_trigger = false;
+      hide_during_completion = false;
+      debounce = 90;
+      keymap = {
+        accept_word = false;
+        accept_line = false;
       };
+    };
   };
 
-  config = mkIf cfg.enable {
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings =
```

Comment:
Didn't we have an example of how to move these out to a separate file recently?

Reply:
Looks like https://github.com/nix-community/nixvim/pull/2647 is the one that had the change

Reply:
Yeah, I forgot about that. Moved it to deprecations.nix